envoy/upstream/resource_manager.h

```diff
@@ -62,6 +62,13 @@ class ResourceManager {
    */
   virtual ResourceLimit& retries() PURE;
 
+  /**
+   * @return Resource& retries that have been scheduled, i.e. the router has decided that
+             the retry will be attempted but has not yet dispatched the retry to a connection pool.
+             This is either due to backoff or due to dispatcher scheduling for an immediate retry.
+   */
+  virtual Resource& retriesScheduled() PURE;
```

Comment:
Need an extra resource tracker for tracking the number of _scheduled_ retries even though there isn't a max. This is because we need them to be counted for the budget resource to use for calculating the max retry concurrency.